instrumentation/apache-httpasyncclient-5.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/apachehttpasyncclient/ApacheHttpAsyncClientTest.java

```diff
@@ -0,0 +1,240 @@
+        String method,
+        URI uri,
+        Map<String, String> headers,
+        RequestResult requestResult) {
+      getClient(uri).execute(request, responseCallback(requestResult));
+    }
+
+    @Override
+    protected void configure(HttpClientTestOptions options) {
+      configureTest(options);
+    }
+  }
+
+  @Nested
+  class ApacheClientHostAbsoluteUriRequestTest extends AbstractHttpClientTest<SimpleHttpRequest> {
+
+    @Override
+    protected SimpleHttpRequest buildRequest(String method, URI uri, Map<String, String> headers) {
+      return configureRequest(new SimpleHttpRequest(method, uri), headers);
+    }
+
+    @Override
+    protected int sendRequest(
+        SimpleHttpRequest request, String method, URI uri, Map<String, String> headers)
+        throws ExecutionException, InterruptedException {
+      return getResponseCode(getClient(uri).execute(request, null).get());
+    }
+
+    @Override
+    protected void sendRequestWithCallback(
+        SimpleHttpRequest request,
+        String method,
+        URI uri,
+        Map<String, String> headers,
+        RequestResult requestResult) {
+      getClient(uri).execute(request, responseCallback(requestResult));
+    }
+
+    @Override
+    protected void configure(HttpClientTestOptions options) {
+      configureTest(options);
+    }
+  }
+
+  SimpleHttpRequest configureRequest(SimpleHttpRequest request, Map<String, String> headers) {
+    request.addHeader("user-agent", "httpasyncclient");
+    headers.forEach((key, value) -> request.setHeader(new BasicHeader(key, value)));
+    return request;
+  }
+
+  int getResponseCode(SimpleHttpResponse response) {
+    return response.getCode();
+  }
+
+  static FutureCallback<SimpleHttpResponse> responseCallback(
+      AbstractHttpClientTest.RequestResult requestResult) {
+    return new FutureCallback<SimpleHttpResponse>() {
+      @Override
+      public void completed(SimpleHttpResponse response) {
+        requestResult.complete(response.getCode());
+      }
+
+      @Override
+      public void failed(Exception e) {
+        requestResult.complete(e);
+      }
+
+      @Override
+      public void cancelled() {
+        requestResult.complete(new CancellationException());
+      }
+    };
+  }
+
+  void configureTest(HttpClientTestOptions options) {
+    options.setUserAgent("httpasyncclient");
+    options.setResponseCodeOnRedirectError(302);
+    options.enableTestReadTimeout();
+    options.setHttpAttributes(
+        endpoint -> {
+          Set<AttributeKey<?>> attributes = new HashSet<>();
+          attributes.add(SemanticAttributes.NET_PEER_NAME);
+          attributes.add(SemanticAttributes.NET_PEER_PORT);
+          attributes.add(SemanticAttributes.HTTP_URL);
+          attributes.add(SemanticAttributes.HTTP_METHOD);
+          attributes.add(SemanticAttributes.HTTP_USER_AGENT);
+          attributes.add(SemanticAttributes.HTTP_SCHEME);
+          attributes.add(SemanticAttributes.HTTP_TARGET);
```

Comment:
`http.scheme` and `http.target` are not captured by the `HttpClientAttributesExtractor`, these are server-only
```suggestion
```